kpi/serializers/v2/asset.py

```diff
@@ -498,18 +498,30 @@ def validate_data_sharing(self, data_sharing: dict) -> dict:
             else:
                 asset = self.instance
                 fields = data_sharing['fields']
-                form_pack, _unused = build_formpack(asset, submission_stream=[])
+                not_supported_types = ['calculate']
+                # We used to get all fields for every version for valid fields,
+                # but the UI shows the latest version only, so only its fields
+                # can be picked up. It is easier then to compare valid fields with
+                # user's choice.
+                form_pack, _unused = build_formpack(
+                    asset, submission_stream=[], use_all_form_versions=False
```

Comment:
OK, we might have to change this in the future, but we'll cross that bridge when we get to it.